qiskit/synthesis/discrete_basis/commutator_decompose.py

```diff
@@ -53,16 +53,42 @@ def _compute_rotation_axis(matrix: np.ndarray) -> np.ndarray:
     """
     _check_is_so3(matrix)
 
+    # If theta represents the rotation angle, then trace = 1 + 2cos(theta).
     trace = _compute_trace_so3(matrix)
-    theta = math.acos(0.5 * (trace - 1))
-    if math.sin(theta) > 1e-10:
-        x = 1 / (2 * math.sin(theta)) * (matrix[2][1] - matrix[1][2])
-        y = 1 / (2 * math.sin(theta)) * (matrix[0][2] - matrix[2][0])
-        z = 1 / (2 * math.sin(theta)) * (matrix[1][0] - matrix[0][1])
-    else:
+
+    if trace >= 3 - 1e-10:
+        # The matrix is the identity (rotation by 0)
         x = 1.0
         y = 0.0
         z = 0.0
+
+    elif trace <= -1 + 1e-10:
+        # The matrix is the 180-degree rotation
+        # We have:
+        #   k_x^2 = (1 + r_00) / 2, k_y^2 = (1 + r_11) / 2, k_z^2 = (1 + r_22) / 2,
+        #   k_x k_y = r_01 / 2, k_y k_z = r_12 / 2, k_x k_z = r_02 / 2
+        squares = [0.5 * (1 + matrix[0][0]), 0.5 * (1 + matrix[1][1]), 0.5 * (1 + matrix[2][2])]
+        index_of_max = squares.index(max(squares))
```

Comment:
There's a function for this 🙂 
```suggestion
        index_of_max = np.argmax(squares)
```

Reply:
nice to know, this is nicer indeed